### ISSUE TITLE
Memory leak due to automatic grouping of MTMD files.

### ISSUE THREAD

**Issue by aowen87** [aowen87/BAR#2076]
Jean Favre reported a memory leak that we suspect is due to automatic file grouping of several MTMD files.

========================================

I need some advice to understand what's causing a big memory leak I see.

I have my own parallel reader for a StructuredGrid. It adds ghostzones (just like the PFLOTRAN and other examples)

Most things work very well in parallel, isosurfaces, slicing, etc. I can animate thru long series of files without any leak.

the only one creating a leak is if I add a

AddPlot("Subset", "curvemesh_257x768x448", 1, 0)
AddOperator("IndexSelect", 0)

to extract a small piece of grid (a solid wall). When I animate, it leaks and then crash.

So I have looked all day at my reader, but isosurfaces and slicing can be animated without leaking. The only one I found leaking is the combo of subset+indexSelect. Does that mean necessarily that the leak is there, or can it be elsewhere? Unfortunately, I have not been able to valgrind it. is it because it is parallel? Seems like valgrind is just ignored. Or else I messed up the setup? I have of course to launch with srun and then valgrind gets in the way. I am not sure how it all gets hooked up together.

========================================
I get the following two regions leaking:

avtDatabasefactory.C (line 880) and
avtMTMDFileFormatinterface.c (line 652)

so this is telling me my reader (or its parent class) is causing problems. The reader is Pixie (my own parallel version), and it is an MTMD thing. On top of that, I use the database grouping to animate timesteps stored in multiple files. Is this a well tested ground?  Is there some confusion because I group a time-series of a reader which itself supports multiple timesteps?

What other reader does that? is there an additional trick to make this go smoothly?
-----------------
Jean

-----------------------REDMINE MIGRATION-----------------------
This ticket was migrated from Redmine. The following information
could not be accurately captured in the new ticket:

Original author: Brad Whitlock
Original creation: 06/06/2012 06:04 pm
Original update: 06/18/2012 06:15 pm
Ticket number: 1089
